include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -1727,6 +1745,36 @@ namespace LinearAlgebra
 
       if (vector_is_ghosted)
         update_ghost_values();
+      else
+        assert_ghost_values_are_zero();
```

Comment:
All of these read funny -- if a vector is ghosted, it makes sense to update the ghost values. But if it *isn't* ghosted, why do we need to check whether the ghost elements are zero? I suspect that what lies behind this question is that you are using the ghost values as *outgoing slots* currently cached locally before being sent elsewhere, but this isn't clear from the function name nor from the if-else statement. Perhaps a different function name and/or a comment in front of these calls would be useful?